octobot_trading/personal_data/portfolios/types/future_portfolio.py

```diff
@@ -33,36 +34,54 @@ def update_portfolio_data_from_order(self, order):
-                order_margin_update_quantity = real_order_quantity / order.filled_price
-                fees_update_quantity = -order.get_total_fees(order.currency)
+    def update_portfolio_data_from_closed_order(self, position, order, realized_pnl_update, size_update, margin_update,
+                                                has_increased_position_size):
+        # when the update comes from a closed order, simultaneously update order and position margin
+        pair_future_contract = order.exchange_manager.exchange.get_pair_future_contract(order.symbol)
+        real_order_quantity = decimal.Decimal(size_update / pair_future_contract.current_leverage).copy_abs()
+
+        # When inverse contract, decrease a currency market equivalent quantity from currency balance
+        if pair_future_contract.is_inverse_contract():
+            order_margin_update_quantity = real_order_quantity / order.filled_price
+            fees_update_quantity = -order.get_total_fees(order.currency)
+            self._update_future_portfolio_data(
+                order.currency,
+                wallet_value=fees_update_quantity + realized_pnl_update,
+                position_margin_value=margin_update,
+                order_margin_value=-order_margin_update_quantity if has_increased_position_size else constants.ZERO)
+
+        # When non-inverse contract, decrease directly market quantity
+        else:
+            # decrease market quantity from market available balance
+            order_margin_update_quantity = real_order_quantity * order.filled_price
+            fees_update_quantity = -order.get_total_fees(order.market)
+            try:
                 self._update_future_portfolio_data(
-                    order.currency,
-                    wallet_value=fees_update_quantity,
-                    position_margin_value=constants.ZERO,
+                    order.market,
+                    wallet_value=fees_update_quantity + realized_pnl_update,    #TODO: take fees into account when full wallet position: take from position (same as pnl)
+                    position_margin_value=margin_update,
                     order_margin_value=-order_margin_update_quantity if has_increased_position_size else constants.ZERO)
 
-            # When non-inverse contract, decrease directly market quantity
-            else:
-                # decrease market quantity from market available balance
-                order_margin_update_quantity = real_order_quantity * order.filled_price
-                fees_update_quantity = -order.get_total_fees(order.market)
+            except errors.PortfolioNegativeValueError:
+                self.logger.warning("Not enough available balance to handle order fees. Reducing position margin...")
+                # part in WIP
+                position.update_size_from_margin(fees_update_quantity if position.is_long() else -fees_update_quantity,
+                                                 update_order_margin=True)
```

Comment:
Why `update_order_margin=True` ?

Reply:
This is in WIP, not sure we really need it yet (I was facing an issue where order fees we creating a negative pf, this might disappear)